engine/src/gldrv/winsys.cpp

```diff
@@ -265,136 +436,136 @@ static bool setup_sdl_video_mode(int *argc, char **argv) {
     if (configuration().graphics.gl_accelerated_visual) {
         SDL_GL_SetAttribute(SDL_GL_ACCELERATED_VISUAL, 1);
     }
-    width = configuration().graphics.resolution_x;
-    height = configuration().graphics.resolution_y;
-
-    // Fix display in fullscreen
-    if(configuration().graphics.full_screen) {
-        // Change base resolution to match screen resolution
-        width = configuration().graphics.resolution_x;//currentDisplayMode.w;
-        height = configuration().graphics.resolution_y;//currentDisplayMode.h;
-        int* ptr_x = const_cast<int*>(&configuration().graphics.bases.max_width);
-        int* ptr_y = const_cast<int*>(&configuration().graphics.bases.max_height);
-        *ptr_x = width;
-        *ptr_y = height;
+    int width = configuration().graphics.resolution_x;
+    int height = configuration().graphics.resolution_y;
+    screen_number = configuration().graphics.screen;
+    bool full_screen = configuration().graphics.full_screen;
+
+    bool result = get_sdl_display_name_by_nr(screen_number, screen_name, instance_ID);
+
+    // width and height are from the config file. We check if this resolution
+    // is supported in full screen mode.
+    if (full_screen) {
+        video_flags |= SDL_WINDOW_OPENGL | SDL_WINDOW_FULLSCREEN | SDL_WINDOW_HIDDEN;
+        int num_modes = 0;
+        bool found = false;
+        SDL_DisplayMode** modes = SDL_GetFullscreenDisplayModes(instance_ID, &num_modes);
+        if (modes) {
+            SDL_DisplayMode* mode = nullptr;
+            for (int i = 0; i < num_modes; ++i) {
+                mode = modes[i];
+                SDL_GetDisplayBounds(instance_ID, &display_bounds);
+                if ((mode->w == width) && (mode->h == height)) {
+                    found = true;
+                    std::memcpy(mode_for_ID, mode, sizeof(SDL_DisplayMode)); // pmx-20251026 I'm not sure of the life length of the data pointed
+                    // by 'mode', for the time being, I prefer to copy. May recheck later.
+                    break;
+                }
+            }
+        }
+        if (found == false) {
```

Comment:
I don't agree with this "rabbit hole". Consider the following scenario:
1. I have a monitor with 1400x1050 (non-standard) and not in the drop-down menu.
2. I manually edit and get it wrong - 1040x1500.
3. The function picks SDL_GetClosestFullscreenDisplayMode 1200x800 (or whatever).
4. I don't know I got it wrong but the display looks a bit off. 
5. I open a bug.

Reply:
I guess that makes sense.

This is how @pmxy had it, but maybe I will change it back.